src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Text.cs

```diff
@@ -157,16 +159,24 @@ internal static void ValidateFormatArgs(string name, CheckTypesContext checkType
             {
                 if (checkTypesContext.Features.PowerFxV1CompatibilityRules)
                 {
-                    var textFormatArgs = new TextFormatArgs
+                    if (string.IsNullOrEmpty(formatArg) || argTypes[0] == DType.UntypedObject || TextFormatUtils.AllowedListToUseFormatString.Contains(argTypes[0]))
                     {
-                        FormatCultureName = null,
-                        FormatArg = null,
-                        HasDateTimeFmt = false,
-                        HasNumericFmt = false
-                    };
+                        var textFormatArgs = new TextFormatArgs
+                        {
+                            FormatCultureName = null,
+                            FormatArg = null,
+                            HasDateTimeFmt = false,
+                            HasNumericFmt = false
+                        };
```

Comment:
Not your changes (it was already there), but this initialization is not needed - the value of textFormatArgs will be set by the call to TextFormatUtils.IsValidFormatArg. You can remove this, and change the call below to `...IsValidFormatArg(formatArg, formatCulture: null, defaultLanguage: null, out var textFormatArgs)`